pkg/resource/stack/secrets.go

```diff
@@ -154,3 +150,35 @@ func (c *cachingCrypter) encryptSecret(secret *resource.Secret, plaintext string
 func (c *cachingCrypter) insert(secret *resource.Secret, plaintext, ciphertext string) {
 	c.cache[secret] = cacheEntry{plaintext, ciphertext}
 }
+
+// mapDecrypter is a Decrypter with a preloaded cache. This decrypter is used specifically for deserialization,
+// where the deserializer is expected to prime the cache by scanning each resource for secrets, then decrypting all
+// of the discovered secrets en masse. Although each call to Decrypt _should_ hit the cache, a mapDecrypter does
+// carry an underlying Decrypter in the event that a secret was missed.
+type mapDecrypter struct {
```

Comment:
Feels like this is similar to the cachingCrypter idea. Wonder if all the caching logic could just be done once, although this caches by cipertext and cachingCrypter is by Secret pointers so maybe makes no sense to combine.

Reply:
> this caches by cipertext and cachingCrypter is by Secret pointers so maybe makes no sense to combine.

This is what led me to keep the two separate. My first instinct was to combine them, but the differences in key type and semantics led me down the proposed approach.

Reply:
Added a comment re: the distinction